library/std/src/lib.rs

```diff
@@ -11,3 +11,33 @@
 
 // re-export all std symbols
 pub use std::*;
+
+/// This assert macro calls kani's assert function passing it down the condition
+/// as well as a message that will be used when reporting the assertion result.
+///
+/// For the first form that does not involve a message, the macro will generate the following message:
+/// assertion "cond"
+/// where `cond` is the stringified condition. For example, for
+/// ```rust
+/// assert!(1 + 1 == 2);
+/// ```
+/// the message will be:
+/// assertion: 1 + 1 == 2
+///
+/// For the second form that involves a custom message possibly with arguments,
+/// the macro will generate a message that is a concat of the custom message
+/// along with all the arguments. For example, for
+/// ```rust
+/// assert!(a + b == c, "The sum of {} and {} is {}", a, b, c);
+/// ```
+/// the assert message will be:
+/// assertion "The sum of {} and {} is {}", 1, 1, 2
```

Comment:
Why not keep it compatible with the std message?

Reply:
You mean remove the `assertion: ` prefix? If so, the reason I added this prefix is to make it easy to match a reachability check with its assert. If not, can you clarify what you mean?

Reply:
The std library adds "assertion failed:" prefix to things that don't have a message. So I was wondering if we should match that.

Not sure I understand how you are planning to match the check with its assert though.

Reply:
Makes sense. The matching should be doable without the `assertion: ` prefix. I'll remove it.